pybamm/models/full_battery_models/equivalent_circuit/thevenin.py

```diff
@@ -0,0 +1,258 @@
+                - "voltage"/"power"/"resistance" : solve an algebraic equation for \
+                    current such that voltage/power/resistance is correct
+                - "differential power"/"differential resistance" : solve a \
+                    differential equation for the power or resistance
+                - "explicit power"/"explicit resistance" : current is defined in terms \
+                    of the voltage such that power/resistance is correct
+                - "CCCV": a special implementation of the common constant-current \
+                    constant-voltage charging protocol, via an ODE for the current
+                - callable : if a callable is given as this option, the function \
+                    defines the residual of an algebraic equation. The applied current \
+                    will be solved for such that the algebraic constraint is satisfied.
+            * "external submodels" : list
+                A list of the submodels that you would like to supply an external
+                variable for instead of solving in PyBaMM. The entries of the lists
+                are strings that correspond to the submodel names in the keys
+                of `self.submodels`.
+    build :  bool, optional
+        Whether to build the model on instantiation. Default is True. Setting this
+        option to False allows users to change any number of the submodels before
+        building the complete model (submodels cannot be changed after the model is
+        built).
+
+    Examples
+    --------
+    >>> import pybamm
+    >>> model = pybamm.equivalent_circuit.Thevenin()
+    >>> model.name
+    'Thevenin Equivalent Circuit Model'
+
+
+    References
+    ----------
+    .. [1] G Barletta, D Piera, and D Papurello. "Thévenin’s Battery Model
+           Parameter Estimation Based on Simulink." Energies 15.17 (2022): 6207.
+    .. [2] N Nieto, L Díaz, J Gastelurrutia, I Alava, F Blanco, JC Ramos, and
+           A Rivas "Thermal modeling of large format lithium-ion cells."
+           Journal of The Electrochemical Society, 160(2), (2012) A212.
+    """
+
+    def __init__(
+        self, name="Thevenin Equivalent Circuit Model", options=None, build=True
+    ):
+        super().__init__(name)
+
+        self.set_options(options)
+        self.param = pybamm.EcmParameters()
+        self.element_counter = 0
+
+        self.set_submodels(build)
+
+    def set_options(self, extra_options=None):
+
+        possible_options = {
+            "calculate discharge energy": ["false", "true"],
+            "operating mode": [
+                "current",
+                "voltage",
+                "power",
+                "differential power",
+                "explicit power",
+                "resistance",
+                "differential resistance",
+                "explicit resistance",
+                "CCCV",
+            ],
+            "number of rc elements": [1, 2, 3, 4, 5, 0],
```

Comment:
Good point there isn't any obvious constraint on this; the model should work but I haven't tested it. I wrote the possible options this way to try to be consistent with the rest of the possible options part of the class (and hoped that no one would want more than 5 😆 )

For the options class, do we have a convention for constraining numeric options? 

Side note, when looking at the class I was thinking that [pydantic](https://pydantic-docs.helpmanual.io/) could help cut the options class code down a lot, make it clearer, and give more flexibility (we could constrain number of RC pairs using `confloat(ge=1)` for example). Might be a bit of work to refactor though so perhaps not worth it.

Reply:
Yes, that's what I was thinking. If it is just for this option probably it's not worth it.

Reply:
We don't have a convention for numeric options but it just needs to pass the test `option in default_options` so you could define a special class whose `in` method returns `True` for integers and `False` otherwise. pydantic looks interesting, always up for hearing about ideas that reduce code complexity

Reply:
The dunder method for `in` is `__contains__`

Reply:
Nice idea! Will give that a go